### PR TITLE
deploy:cleanup should skip only the invalid releases and continue the rotation of releases

### PR DESCRIPTION
**Pull request by darwin67**
### Summary
https://github.com/capistrano/capistrano/pull/1883 seems to be stalled so creating a new PR.
This change will result in skipping invalid release format directories, but will still continue on the rotation of the valid ones.

### Short checklist

- [x] Did you run `bundle exec rubocop -a` to fix linter issues?
- [x] Did you confirm that the RSpec tests pass?
- [x] If you are fixing a bug or introducing a new feature, did you add a CHANGELOG entry?

### Other Information
Don't see any tests for the rake tasks so if there's a certain format you want me to write it, or if you could point me in the right direction. I'm totally happy to add specs for the change.

### CODE REVIEW

**Comment by mattbrictson** [comment]
This PR is much appreciated; thanks for your patience! I will try to find time to review and provide guidance on the testing strategy this week.

**Comment by darwin67** [comment]
Thank you for the feedback.
I'll see if I can get to it this week.

**Comment by darwin67** [comment]
@mattbrictson 
- updated as suggested
- added cucumber specs for `deploy:cleanup`

## Addition
- I added ruby `2.4.1` to travis since it works runs fine on my machine.
- vagrant seems to work fine with the latest bundler `v1.15.3` so I deleted the requirements for `1.10.5` in `DEVELOPMENT.md`
```
# machine environment
Ubuntu 16.04 LTS
ruby - 2.4.1
bundler - 1.15.3
vagrant - 1.9.7
virtualbox - 5.1.24 r117012 (Qt5.5.1)
```

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 language: ruby
 rvm:
+  - 2.4.1
```
Comment (mattbrictson) [inline .travis.yml]: Thanks 👍

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -28,11 +28,10 @@ Capistrano is a Ruby project, so we expect you to have a functioning Ruby enviro
 
 Make sure to install:
 
-* [Bundler](https://bundler.io/) 1.10.5 (see note)
+* [Bundler](https://bundler.io/)
 * [Vagrant](https://www.vagrantup.com/)
 * [VirtualBox](https://www.virtualbox.org/wiki/Downloads) (or another [Vagrant-supported](https://docs.vagrantup.com/v2/getting-started/providers.html) VM host)
 
-*Note: As of this writing (December 2015), Vagrant does not work with Bundler > 1.10.5. If you have multiple versions of Bundler installed, use the special underscore command-line argument to force a compatible version, like this: `bundle _1.10.5_ exec rake features`.*
```
Comment (mattbrictson) [inline DEVELOPMENT.md]: Good find! Excellent ✨

File: features/deploy.feature [hunk 3]
```diff
@@ -52,6 +52,14 @@ Feature: Deploy
     When I run cap "deploy:symlink:release"
     Then the current directory will be a symlink to the release
 
+  Scenario: Cleanup
```
Comment (mattbrictson) [inline features/deploy.feature]: This is a great test. 💯

**Pull request merged**